api/web3server.go

```diff
@@ -831,31 +830,33 @@ func (svr *Web3Server) getFilterChanges(in interface{}) (interface{}, error) {
 	filterID = removeHexPrefix(filterID)
 	filterObj, err := loadFilterFromCache(svr.cache, filterID)
 	if err != nil {
-		return []interface{}{}, err
+		return nil, err
```

Comment:
return nil is OK?
`nil` may cause panic if trying to json-decode it, but `[]interface{}{}` will not

Reply:
no. `func packAPIResult()` will only encode err